iroh/src/node.rs

```diff
@@ -728,4 +736,72 @@ mod tests {
         println!("addrs: {:?}", ticket.addrs());
         assert!(!ticket.addrs().is_empty());
     }
+
+    #[tokio::test]
+    async fn test_node_add_collection_event() {
+        let db = Database::from(HashMap::new());
+        let node = Builder::with_db(db)
+            .bind_addr((Ipv4Addr::UNSPECIFIED, 0).into())
+            .runtime(&test_runtime())
+            .spawn()
+            .await
+            .unwrap();
+
+        let _drop_guard = node.cancel_token().drop_guard();
+
+        let mut events = node.subscribe();
+        let (tx, rx) = tokio::sync::oneshot::channel();
+        tokio::spawn(async move {
+            let mut got_hash = None;
+            while let Ok(msg) = events.recv().await {
+                match msg {
+                    Event::ByteProvide(e) => {
+                        if let iroh_bytes::provider::Event::CollectionAdded { hash } = e {
+                            got_hash = Some(hash);
+                            break;
+                        }
+                    }
+                }
+            }
```

Comment:
This test could be expanded in the future for more extensive provider event testing